packages/datadog-instrumentations/src/elasticsearch.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const startCh = channel('apm:elasticsearch:query:start')
+const asyncEndCh = channel('apm:elasticsearch:query:async-end')
+const endCh = channel('apm:elasticsearch:query:end')
+const errorCh = channel('apm:elasticsearch:query:error')
+
+addHook({ name: '@elastic/elasticsearch', file: 'lib/Transport.js', versions: ['>=5.6.16'] }, Transport => {
+  shimmer.wrap(Transport.prototype, 'request', wrapRequest)
+  return Transport
+})
+
+addHook({ name: 'elasticsearch', file: 'src/lib/transport.js', versions: ['>=10'] }, Transport => {
+  shimmer.wrap(Transport.prototype, 'request', wrapRequest)
+  return Transport
+})
+
+function wrapRequest (request) {
+  return function (params, options, cb) {
+    if (!startCh.hasSubscribers) {
+      return request.apply(this, arguments)
+    }
+
+    if (!params) return request.apply(this, arguments)
+
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
```

Comment:
Maybe we should start giving these better names, i.e. relating to whatever is actually happening. No need to just copy what `bind` is doing internally. @rochdev what do you think?

Reply:
This only impacts the name of the type received by the `init` hook, so we should put whatever we want the hook to receive. I don't have a strong opinion on what that should be, especially since these hooks are likely to eventually be deprecated anyway so the type name will become basically useless information.

Reply:
I think it's better to have it make some sense if it's going to be visible to other tooling.

But no need to block on it so I'll call this resolved.